connectors/src/connectors/slack/bot.ts

```diff
@@ -189,13 +190,59 @@ async function botAnswerMessage(
       }
     }
   }
+  // Extract all ~mentions.
+  const mentionCandidates = message.match(/~(\S+)/g) || [];
+
+  let mentions: { assistantName: string; assistantId: string }[] = [];
+  if (mentionCandidates.length > 0) {
+    const agentConfigurationsRes = await dustAPI.getAgentConfigurations();
+    if (agentConfigurationsRes.isErr()) {
+      return new Err(new Error(agentConfigurationsRes.error.message));
+    }
+    const agentConfigurations = agentConfigurationsRes.value;
+    for (const mc of mentionCandidates) {
+      const scores: {
+        assistantId: string;
+        assistantName: string;
+        distance: number;
+      }[] = [];
+      for (const agentConfiguration of agentConfigurations) {
+        const distance = editDistance(
+          mc.slice(1).toLocaleLowerCase(),
+          agentConfiguration.name.toLowerCase()
+        );
+        scores.push({
+          assistantId: agentConfiguration.sId,
+          assistantName: agentConfiguration.name,
+          distance: distance,
+        });
+      }
+      scores.sort((a, b) => {
```

Comment:
nit: maybe no need to sort the whole array if we only need the max ?

Reply:
stored the max. thanks.